### PR TITLE
Improve tests

### PR DESCRIPTION
**Pull request by marcrowo**
- Slight improvement to the very simple variation test.
- Added a new test for the `telemetry()` function. There is test is a little hacky because of a strange timing bug with Telemetry. This seems like a bug in the Telemetry library, not the shield-study-utils library.
- Added test for `startup()` process and `endStudy()` function.

Note: `test-addon/StudyUtils.jsm` is now being ignored, which causes a large diff.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,4 +6,3 @@ test/z-ensure-coverage.js
 
 *.*.swp
 dist/
-examples/stage-2-shield-study/addon/StudyUtils.jsm
```
Comment (gregglind) [inline .gitignore]: I don't understand this removal. It's a built file.

File: src/StudyUtils.in.jsm [hunk 2]
```diff
@@ -335,7 +335,7 @@ class StudyUtils {
     return queryArgs;
   }
 
-  _telemetry(data, bucket = "shield-study-addon") {
+  async _telemetry(data, bucket = "shield-study-addon") {
```
Comment (gregglind) [inline src/StudyUtils.in.jsm]: This is fine.  Wasn't sure if it had a useful return value or not!  EVERYTHING with Telemetry ID is async, fwiw.

File: test-addon/utils.jsm [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      }},
+    addon: {id: "1", version: "1"},
+    telemetry: { send: true, removeTestingFlag: false },
+  });
+  studyUtils.setVariation({ name: "puppers", weight: "2" });
+}
+
+// Returns array of pings of type `type` in sorted order by timestamp
+// first element is most recent ping
+async function getMostRecentPingsByType(type) {
```
Comment (gregglind) [inline test-addon/utils.jsm]: this is awesome!

File: .gitignore (hunk 1 above)
Comment (marcrowo) [inline .gitignore]: This was moved to a [.gitignore](https://github.com/mozilla/shield-studies-addon-utils/pull/78/files#diff-2054ac9a8ccbc631331888139d01b03c) within the examples folder [as per our previous discussion](https://github.com/mozilla/shield-studies-addon-utils/pull/77#discussion_r128598424).

**Pull request merged**